subprojects/core-api/src/main/java/org/gradle/api/file/BuildLayout.java

```diff
@@ -33,7 +33,7 @@
  * @since 8.5
  */
 @Incubating
-@ServiceScope(Scope.Build.class)
+@ServiceScope(Scope.Settings.class)
 public interface BuildLayout {
```

Comment:
ℹ️  The `BuildLayout` was incorrectly marked with a `Build` scope. However, since it is actually a `Settings` scope service, it is not available for injection in the `Project` scope.

Reply:
🤔 I think this might be another code smell, indicating that the names of our scopes are misaligned. For me it doesn't make sense that `BuildLayout` is not available in `Build` scope; and we do use it in evaluating projects, so why is it not available in `Project` scope?

Reply:
I wouldn't call it a code smell. It was a design choice at the time of the implementation. cc @abstratt

> and we do use it in evaluating projects

I don't believe that's the case currently.